### ISSUE TITLE
[dreambooth] RuntimeError when trying to Train on a Tesla T4 on Colab.

### ISSUE THREAD

**Issue by Maki9009** [huggingface/diffusers#668]
### Describe the bug

just right when it starts running the training it crashes with out of memory error. When i set it to fp16, i get a unscale_() was already called for error.. I'm not sure how to run this on a 16gb GPU? 

### Reproduction

Tesla T4, using 12 training images and i set the step count 1200 and max 100 regulation images. 

### Logs

```shell
RuntimeError                              Traceback (most recent call last)
<ipython-input-24-c6e3ce5f5a40> in <module>
      1 #@title Run training
      2 import accelerate
----> 3 accelerate.notebook_launcher(training_function, args=(text_encoder, vae, unet))
      4 with torch.no_grad():
      5     torch.cuda.empty_cache()

4 frames
/usr/local/lib/python3.7/dist-packages/torch/cuda/amp/grad_scaler.py in unscale_(self, optimizer)
    268 
    269         if optimizer_state["stage"] is OptState.UNSCALED:
--> 270             raise RuntimeError("unscale_() has already been called on this optimizer since the last update().")
    271         elif optimizer_state["stage"] is OptState.STEPPED:
    272             raise RuntimeError("unscale_() is being called after step().")

RuntimeError: unscale_() has already been called on this optimizer since the last update().
```

### System Info

DreamBooth Diffuser.

**Comment by patrickvonplaten** [comment]
Heey @Maki9009,

Thanks for raising the issue. Gently pinging @patil-suraj here

**Comment by patil-suraj** [comment]
Thanks for the issue, did you use `mixed_precision` for training ?

**Comment by patil-suraj** [comment]
Can reproduce the issue. Note that this is not an OOM but an issue with gradient clipping when doing gradient accumulation.
fix is here #686
